cmd/kubeadm/app/phases/upgrade/compute.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"fmt"
+)
+
+// Upgrade defines an upgrade possibility to upgrade from a current version to a new one
+type Upgrade struct {
+	Description string
+	Before      StateBeforeUpgrade
```

Comment:
> we could just make it a list of one element

I'll do that. Basically, this was to make the output look like this:
```
COMPONENT   CURRENT      AVAILABLE
Kubelet     1 x v1.7.0   v1.7.3
            1 x v1.7.1   v1.7.3
            1 x v1.7.2   v1.7.3
```

(happy to revisit that part as well)